### PR TITLE
[CentOS 8] Dont patch any files and start building ninja

### PR DESCRIPTION
**Pull request by shahmishal**

### CODE REVIEW

**Comment by tomerd** [comment]
@shahmishal #30 haha :D

**Comment by tomerd** [comment]
there is some overlap but also looks like you were able to get rid from more patches. so either we combine into one or we merge one then rebase the other

File: platforms/Linux/centos/8/swift-lang.spec [hunk 1]
```diff
@@ -5,6 +5,7 @@
 %global yams_version 4.0.2
 %global sap_version 0.4.3
 %global swift_crypto_version 1.1.5
+%global ninja_version v1.10.2
```
Comment (tomerd) [inline platforms/Linux/centos/8/swift-lang.spec:8]: I think this needs to be 1.10.2

File: platforms/Linux/centos/8/swift-lang.spec [hunk 2]
```diff
@@ -32,10 +33,7 @@ Source15:       https://github.com/unicode-org/icu/archive/release-%{icu_version
 Source16:       https://github.com/apple/swift-syntax/archive/swift-%{swifttag}.zip#/swift-syntax.tar.gz
 Source17:       https://github.com/jpsim/Yams/archive/%{yams_version}.zip
 Source18:       https://github.com/apple/swift-crypto/archive/refs/tags/%{swift_crypto_version}.tar.gz
-
-Patch0:         patches/build-presets.patch
-Patch1:         patches/swift-api-checker.patch
-Patch2:         patches/hwasan_symbolize.patch
+Source19:       https://github.com/ninja-build/ninja/archive/refs/tags/%{ninja_version}.tar.gz
```
Comment (tomerd) [inline platforms/Linux/centos/8/swift-lang.spec:36]: add the `v` prefix here

**Comment by shahmishal** [comment]
> @shahmishal #30 haha :D

Haha, I thought we talked offline about this yesterday. I might have miss understood who is working on it.

File: platforms/Linux/centos/8/swift-lang.spec [hunk 3]
```diff
@@ -124,14 +121,8 @@ mv icu-release-%{icu_version} icu
...
-# Adjust presets
-%patch0 -p0
-
-# Adjust python version swift-api-checker
-%patch1 -p1
-
-# Adjust python version hwasan_symbolize
-%patch2 -p1
+# Ninja
+mv %{ninja_version} ninja
```
Comment (tomerd) [inline platforms/Linux/centos/8/swift-lang.spec:125]:
```suggestion
mv ninja-%{ninja_version} ninja
```

**Comment by tomerd** [comment]
> Haha, I thought we talked offline about this yesterday. I might have miss understood who is working on it.

no worried let take your and we can rebase mine with the additional fixes for tarball names on top

**Comment by shahmishal** [comment]
> > Haha, I thought we talked offline about this yesterday. I might have miss understood who is working on it.
> 
> no worried lets take your and we can rebase mine with the additional fixes for tarball names on top

Open to either PR :)

**Comment by tomerd** [comment]
> Open to either PR :)

lets start with ours, added some comments inline

File: platforms/Linux/centos/8/Dockerfile [hunk 4]
```diff
@@ -22,9 +22,6 @@ RUN yum-builddep -y /root/rpmbuild/SPECS/swift-lang.spec
 # Get the sources for Swift as defined in the spec file
 RUN spectool -g -R /root/rpmbuild/SPECS/swift-lang.spec
 
-# Add the patches
-ADD patches/*.patch /root/rpmbuild/SOURCES/
```
Comment (tomerd) [inline platforms/Linux/centos/8/Dockerfile:26]: keep this in case we have patches in the future

Comment (shahmishal) [reply platforms/Linux/centos/8/Dockerfile:26]: We should avoid patches, and have this resolved before release.

Comment (tomerd) [reply platforms/Linux/centos/8/Dockerfile:26]: sure we can add it back if we need it

**Review by tomerd — APPROVED** [review]

**Comment by shahmishal** [comment]
We still need two patches: 
* hwasan_symbolize.patch
* swift-api-checker.patch

This will fix the Swift patch - https://github.com/apple/swift/pull/39614

**Pull request closed without merging**